### DISCUSSION TITLE
Inline images are not rewritten by Mercury extractor and remain pointing to their original URLs

### DISCUSSION THREAD

**Discussion by mtvu** [ArchiveBox/ArchiveBox#770] [Q&A] [answered]
I notice the content.html under mercury directory still contains links to online images. Does that mean the source  images could not be displayed if they are removed

![image](https://user-images.githubusercontent.com/4926855/121780417-8a1b6d00-cbe3-11eb-891c-08c369a2b33b.png)

miro.medium.com directory contains downloaded images from the archived website. How does it get used ?

**Comment by pirate** [comment, marked as answer]
I believe mercury does not have an option to download/inline the images like Singlefile does. I recommend opening a feature request issue on the [Mercury repo](https://github.com/postlight/mercury-parser/issues/) to ask them to add an option to inline images as base64-encoded `data:...`  urls (like how SingleFile does it).

If they don't seem interested in adding that (see https://github.com/postlight/mercury-parser/issues/438 for past precedent), then as a last-resort we could try to add a post-processing step after we run mercury to rewrite the image urls to relative URLs or `data:...` inlines, but I'm reluctant as it's a lot of extra code for us to maintain and will likely be buggy as URL rewriting and image recoding is a hard problem.
